doc: ncbi datasets

j23414 opened the pull request:
## Description of proposed changes

Since the [transition from NCBI Virus to NCBI datasets]( https://github.com/nextstrain/monkeypox/commit/82ace30abd07bec140c7e14aee29a46c7bfb28a8), revised the documentation to reflect changes in how metadata and sequences are fetched from NCBI.

Feel free to propose alternative wording or descriptions.

## Related issue(s)

* https://github.com/nextstrain/monkeypox/commit/82ace30abd07bec140c7e14aee29a46c7bfb28a8
* https://github.com/nextstrain/dengue/pull/13#pullrequestreview-1675235743

## Checklist

* [x] Checks pass

joverlee521 reviewed (changes requested):
Thanks for updating the docs here! (I should have done this as part of https://github.com/nextstrain/monkeypox/pull/179.) 

I recommend adding a "Software requirements" section (similar to the [phylogenetic/README](https://github.com/nextstrain/monkeypox/tree/master/phylogenetic#software-requirements)) that points to the centralized Nextstrain installation docs instead of adding detailed docs on the NCBI Datasets here. 

NCBI Datasets CLI is included in the docker/conda runtimes and would not need to be installed separately. Once we add `ncbi-datasets-cli` to our ambient installation instructions, the centralized installation docs should cover everything. Then we wouldn't need to maintain separate docs within each pathogen repo that uses NCBI Datasets.

j23414 commented:
Thanks @joverlee521! Updated centralized Nextstrain installation docs in https://github.com/nextstrain/docs.nextstrain.org/pull/172 and added a software requirments section. This PR is ready for review.
